### PR TITLE
learner prerequisites 

### PR DESCRIPTION
**Pull request by psteinb**
This file lists questions that can be included in a pre-workshop survey in order for instructors to get to know their learners **BEFORE** as workshp. I'd be curious for reviews.

Note: At best, this document should be referenced in some kind of `instructors guide`. But if I am not mistaken, such a guide doesn't exist.

### CODE REVIEW

File: _extras/learner-prerequisites.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+This page collects questions that learners can answer in order to judge if they 
+bring all prerequisites to the course.
+
+# Required Pre-Knowledge
+
+- **Basic Shell** – Navigating directories, Copy/Moving, writing shell scripts, 
+using the environment, using wildcards
+
+# Pre-Workshop Survey
```
Comment (tkphd) [inline _extras/learner-prerequisites.md:13]: A proper Markdown file contains only one top-level header (`#`). Consider using `{: .prereq}` blocks for these two?

**Comment by tkphd** [comment]
Would it make more sense to provide this checklist in `setup.md`, where learners are directed to visit ahead of the workshop already, instead of a separate page the must navigate to?

File: _extras/learner-prerequisites.md (hunk 1 above)
Comment (psteinb) [reply _extras/learner-prerequisites.md]: see 80ea80c

**Comment by psteinb** [comment]
that is something to discuss. My contribution was intended to provide something like a catalogue from which people can select and reuse questions to their own liking.

If you think that a self-organized survey on `setup.md`, then I suggest to create a subsequent PR in which we select the questions that are suitable.

File: _extras/learner-prerequisites.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+2. I'd need to look up the syntax in a cheatsheet or some old code and I'm good to do this.
+3. I am unclear about this, I'd have to consult a colleague or a search engine to do this.
+4. I am not sure what to do.
+
+#### Automating All The Things
+
+You observe yourself copy and pasting the same sequence of 5 commands on the command line for more 
+than three times during a day. You alert yourself that you should write a script 
+for this now. The script requires two input arguments, i.e. two single file locations, 
+to perform it's duty.
```
Comment (mikerenfro) [inline _extras/learner-prerequisites.md:60]: `its`, with no apostrophe.

**Review by mikerenfro — CHANGES_REQUESTED** [review]
All I see is this one change. The rest of the conversation is up to y'all.

File: _extras/learner-prerequisites.md (hunk 2 above)
Comment (psteinb) [reply _extras/learner-prerequisites.md]: great find! thanks. see 3567b20

File: _extras/learner-prerequisites.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Required Pre-Knowledge
+
+> **Basic Shell** – Navigating directories, Copy/Moving, writing shell scripts, 
+> using the environment, using wildcards
+{: .prereq}
+
+## Pre-Workshop Survey
+
+For a motivation of this survey type, see [Greg Wilsons Template in Teaching
```
Comment (tkphd) [inline _extras/learner-prerequisites.md:16]: Wilson's

File: _extras/learner-prerequisites.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+## Pre-Workshop Survey
+
+For a motivation of this survey type, see [Greg Wilsons Template in Teaching 
+Tech Together](https://teachtogether.tech/en/index.html#s:checklists-preassess).
+
+### Shell
+
+#### Moving Things
+
+You are provided with a directory of `300` files that end with `.log`, `.data`
```
Comment (tkphd) [inline _extras/learner-prerequisites.md:23]: 300 is just a number, not code -- drop backticks

File: _extras/learner-prerequisites.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+For a motivation of this survey type, see [Greg Wilsons Template in Teaching 
+Tech Together](https://teachtogether.tech/en/index.html#s:checklists-preassess).
+
+### Shell
+
+#### Moving Things
+
+You are provided with a directory of `300` files that end with `.log`, `.data` 
+and `.err` at equal proportions. You like to rename all `.log` files to `.out`
```
Comment (tkphd) [inline _extras/learner-prerequisites.md:24]: "You want to" or "You would like to"

File: _extras/learner-prerequisites.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+3. I am unclear about this, I'd have to consult a colleague or a search engine to do this.
+4. I am not sure what to do.
+
+#### Collaborator Candy
+
+A collaborator provides you an implementation of a state-of-the-art simulation 
+that you need to compare your own predictions to. He tells you: "You can use it on the command line 
+right away. Unpack the file I sent you and use the `sim` executable in the 
+`bin/` folder from it. The rest is explained in the output of the `--help` flag." 
+You want to execute `sim` on your cluster. How do you set up your shell for this?
```
Comment (tkphd) [inline _extras/learner-prerequisites.md:48]: "How do you setup your shell" is confusing -- perhaps "What steps would you take to get the `help` message printed in your shell?"

File: _extras/learner-prerequisites.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+1. I can do that. Give me a shell and I'll show you.
+2. I'd need to look up the syntax in a cheatsheet or some old code and I'm good to do this.
+3. I am unclear about this, I'd have to consult a colleague or a search engine to do this.
+4. I am not sure what to do.
+
+#### Automating All The Things
+
+You observe yourself copy and pasting the same sequence of 5 commands on the command line for more
```
Comment (tkphd) [inline _extras/learner-prerequisites.md:57]: Strange third-person observer phrasing. Consider

"You notice that you've been copying & pasting the same sequence of 5 shell commands more than a few times during the day. It occurs to you that capturing the workflow in a shell script would simplify the task and make it more repeatable," etc.

**Review by tkphd — CHANGES_REQUESTED** [review]
Very good. Requesting some grammar fixes.

File: _extras/learner-prerequisites.md (hunk 3 above)
Comment (tkphd) [reply _extras/learner-prerequisites.md]: Fixed in 7da0806

File: _extras/learner-prerequisites.md (hunk 4 above)
Comment (tkphd) [reply _extras/learner-prerequisites.md]: Fixed in 7da0806

File: _extras/learner-prerequisites.md (hunk 5 above)
Comment (tkphd) [reply _extras/learner-prerequisites.md]: Fixed in 7da0806

File: _extras/learner-prerequisites.md (hunk 6 above)
Comment (tkphd) [reply _extras/learner-prerequisites.md]: Fixed in 7da0806

File: _extras/learner-prerequisites.md (hunk 7 above)
Comment (tkphd) [reply _extras/learner-prerequisites.md]: Fixed in 7da0806

**Comment by tkphd** [comment]
@psteinb I made edits to address my review suggestions. Does it look OK to you?
@mikerenfro your request for changes is blocking, has your concern been addressed?

**Review by mikerenfro — APPROVED** [review]
I thought I'd resolved my one thing and released the hold. Apparently not, so hopefully this does it.

**Review by mikerenfro — APPROVED** [review]
Order of operations, yo. Too many buttons to click in the wrong order.

**Pull request merged**